src/NzbDrone.Core/HealthCheck/Checks/ApiKeyValidationCheck.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System.Text.RegularExpressions;
+using NLog;
+using NzbDrone.Core.Configuration;
+using NzbDrone.Core.Configuration.Events;
+using NzbDrone.Core.Lifecycle;
+
+namespace NzbDrone.Core.HealthCheck.Checks
+{
+    [CheckOn(typeof(ApplicationStartedEvent))]
+    [CheckOn(typeof(ConfigSavedEvent))]
+    public class ApiKeyValidationCheck : HealthCheckBase
+    {
+        private static readonly Regex ValidateApiKeyRegex = new ("^[a-zA-Z0-9]{20,}$", RegexOptions.Compiled);
```

Comment:
Since we’re just going to throw an health check back telling them to choose a more secure API key, should we instead just check for a minimum length and avoid a specific format? We’ll now handle anything they throw at us, but we should expect a minimum length.

This will also avoid us calling out a more complex 20 character token as insecure.